fe/fe-core/src/main/java/com/starrocks/authentication/UserProperty.java

```diff
@@ -15,66 +15,319 @@
 
 package com.starrocks.authentication;
 
+import com.google.common.collect.Lists;
 import com.google.gson.annotations.SerializedName;
+import com.starrocks.analysis.StringLiteral;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.InternalCatalog;
 import com.starrocks.common.Config;
 import com.starrocks.common.DdlException;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
 import com.starrocks.common.Pair;
-import com.starrocks.sql.ast.SetUserPropertyVar;
+import com.starrocks.connector.exception.StarRocksConnectorException;
+import com.starrocks.privilege.AccessDeniedException;
+import com.starrocks.privilege.AuthorizationMgr;
+import com.starrocks.privilege.ObjectType;
+import com.starrocks.privilege.PrivilegeException;
+import com.starrocks.privilege.PrivilegeType;
+import com.starrocks.qe.VariableMgr;
+import com.starrocks.server.CatalogMgr;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.server.MetadataMgr;
+import com.starrocks.sql.analyzer.Authorizer;
+import com.starrocks.sql.ast.SystemVariable;
+import com.starrocks.sql.ast.UserIdentity;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
+import java.lang.reflect.Field;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
 
+// UserProperty is a class that represents the properties that are identified.
 public class UserProperty {
+    private static final Logger LOG = LogManager.getLogger(UserProperty.class);
+
+    public static final String PROP_MAX_USER_CONNECTIONS = "max_user_connections";
+    public static final String PROP_CATALOG = "catalog";
+    public static final String PROP_DATABASE = "database";
+    public static final String PROP_SESSION_PREFIX = "session.";
+
+    public static final long MAX_CONN_DEFAULT_VALUE = 1024;
+    public static final String CATALOG_DEFAULT_VALUE = InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME;
+    public static final String DATABASE_DEFAULT_VALUE = "";
+    public static final String EMPTY_VALUE = "";
+
     @SerializedName(value = "m")
-    private long maxConn = 1024;
+    private long maxConn = MAX_CONN_DEFAULT_VALUE;
```

Comment:
Why do we need to manage these separately? Can we unify the coding with system variables? Just make sure that user properties and system variables do not have the same name (this is easy for us to do because user properties are also under our control). The current approach 1. It brings complexity to the code. 2. The logic is not unified. There is a catalog in the user property and there is also a catalog in the session variable, and it is called "session.catalog". How to maintain the relationship between these two pieces of data, and what is the meaning of these two identical pieces of data? We should use user properties as a supplement and overwrite of system variables.